Scripts/Runtime/Logging/.CSProject/UnityLogListener.cs

```diff
@@ -267,47 +299,52 @@ private void ProcessPendingLogs()
             m_IsHandlingBurstedLog = false;
         }
 
-
-
-        private void SendToLogger(UnityEngine.Object context, LogLevel logLevel, string message)
+        private void SendToLogger(UnityEngine.Object context, LogLevel logLevel, string message, StackTrace stackTrace = null)
         {
-            // Skip 4 frames to get to the original caller
-            // 0 - Here
-            // 1 - LogException or LogFormat
-            // 2 - UnityEngine.Logger.Log(+Warning, +Error, ...), Assert
-            // 3 - UnityEngine.Debug.Log(+Warning, +Error, ...), Assert
-            // 4 - Caller of Debug.Log(+Warning, +Error, ...), Assert
-            (StackFrame callerFrame, MethodBase callerMethod) = ResolveCaller(4);
+            (StackFrame callerFrame, MethodBase callerMethod) = ResolveCaller(stackTrace);
+
+            string callerPath = callerFrame?.GetFileName() ?? callerMethod?.ReflectedType.FullName;
```

Comment:
I changed this fallback to `ReflectedType.FullName` instead of just `Name` because the type name alone in library code was proving insufficient context.

However, I'm only seeing the namespace in the console, because `UnityLogHandler` is expecting an absolute file path and running `Path.GetFileNameWithoutExtension(...)` on it, which trims the "extension" (type name) off the end :P

I could move that up, but this leads to a greater discussion - should `callerPath` be more generic, like `callerName` or something? Even in the default case, would the `method.ReflectedType.Name` be more useful than the `frame.GetFileName()` even though those are typically the same?

Reply:
The problem with changing this higher up in the `Logger` is that the normal resolution of the file name is resolved for free at compile time using the `[CallerFilePath]` parameter attribute. I wouldn't want to give that up as it represents the vast majority of logging calls.

What if we constructed a fake path based on the reflected type name?
Ex: If you need to use the reflected type you could pass in a "file path" of
`<Virtual Runtime>/{fullTypeMame}.generated`

maybe even replace `.` in the namespace with `/` or group the whole namespace into a single directory name. Ex: `<Virtual Runtime>/My.Name.Space/MyClass.generated`


If the type name on its own isn't enough the log handler(s) that you're using should be configurable to prepend different/more information to a log message if needed for a specific use case.

Reply:
The fake "type path" seems the most functional, but boy what a code smell. I might simplify things and leave the fallback as `Name` instead of `FullName` - in some cases it's weirdly abstract like `(Debug|...` but other times `(Anvil.Unity.Logging.UnityLogListener|...` is too much.

For what it's worth, I tried checking for null/empty `callerPath` in the handler instead and falling back on `callerDerivedTypeName` instead (i.e. the owner of the Logger instance) but that doesn't always match the log context - for example, if I create a second instance of `UnityLogListener` in some custom script, I want the log context to be `UnityLogListener` as that's what threw the exception, but Unity internally calls `ILogListener.LogException` with the `context` object set to that custom script, so the Logger instance ends up belonging to that.

Anyway, I think simply falling back on `DeclaringType.Name` is simple and easy, and there are ways to get more context if you need it.

Reply:
Here's a thought - if the fallback fails too, should there be some default text instead of just blank? For example, if the entire call stack happens to use `[Exclude]` attributes, instead of the context looking like `(|:0)` it could be `(<unknown>|<unknown>:0)` or similar?

Also, is it more readable for the line number to be grouped with the file name instead?
```
(MyScript:42|OnEnable) Enabling my object!
```

Reply:
Fallback:

Yeah it would be nice to see that the fallback failed in the log. When I see `(|:0)` it makes me think that there was some weirdness with the execution and what i was trying to log was hidden in some DLL or I didn't have permissions or whatever. It looks like a glitch rather than something I can investigate and fix. 

`(<file-failed>:0|<function-failed>)` would probably prompt me to start digging. 

Readability:

Yes! I like that. `(FILE:LINE|FUNCTION) MESSAGE` makes sense to me.

Reply:
Unqualified name:
Yea, don't use fully qualified name. Most of the time you can quickly figure out any type name ambiguity based on the context of the log message or a quick check of the types with the same name. Prefer easier to scan logs over saving the odd bit of time wasted checking multiple types with the same name.

Also, in Unity you can just check the rest of the stack attached to the message and get exactly what you want.

Fallback:
Agreed that having some placeholder text is a good idea. I think I prefer `<unknown>` but I'm not picky on this.

Readability:
Disagree on the default changing...not just because I came up with it 😆 
Right now it reads widest scope to narrowest scope. If people prefer something different on their projects they can change the formatting on the handlers for their specific application.

I vote it stays as is but it's not just up to me!

Reply:
Readability: I don't feel too strongly on this to cast a serious vote. Widest to narrowest scope is a valid argument.

Reply:
My only argument for changing the log context order is that I didn't realize until this PR that the first part was the file name - I assumed it was the Logger's owner, or at least the type that declares the method. I suggested `(File:Line|Method)` because `File:Line` is a familiar format, and makes it more obvious that it's saying "this file at this line". (If I could figure out how, I would skip the line number when the file path falls back on `DeclaringType.Name` but that's a whole other issue).

If instead the goal is to sort by scope, I would trade the bar for a dot: `(File.Method:Line)` so they don't look like separate pieces, otherwise I read it as ("file/type name" | "method at line #")

If you're still not convinced, I'm fine leaving it as-is, just wanted to ask the question and share my rationale!

Reply:
Your rationale is valid. Feel free to change the handlers to a different default format. Unless you've already done the work let's submit the change in a separate PR though. (#79)

> (If I could figure out how, I would skip the line number when the file path falls back on DeclaringType.Name but that's a whole other issue).

Not really possible today but you could configure to always include both in your application if you wanted. Probably not ideal.

> If instead the goal is to sort by scope, I would trade the bar for a dot: (File.Method:Line) so they don't look like separate pieces, otherwise I read it as ("file/type name" | "method at line #")

The trouble with a `(File.Method:Line)` is that it's misleading since the file name is not necessarily close to the type name. I lived with this for a bit in a project and it wasn't nice.